File: Semester4/Systems for Design and Implementation/mpp-proiect-csharp-raulfrandes/services/IServices.cs
using System.Collections.Generic;
using model;
using model.DTOs;

namespace services;

public interface IServices
{
    Referee Login(string username, IObserver client);
    List<TrialDTO> GetTrialNames();
    List<ParticipantDTO> GetParticipantsAndPoints();
    List<ParticipantDTO> FilterParticipantsByTrial(long idTrial);
    void AddResult(Result result);
    void Logout(long idReferee, IObserver client);
    void SendFilter(long idTrial);
}

File: Semester4/Systems for Design and Implementation/mpp-proiect-java-raulfrandes/TriathlonServices/src/main/java/triathlon/services/IServices.java
package triathlon.services;


import triathlon.model.DTOs.ParticipantDTO;
import triathlon.model.DTOs.TrialDTO;
import triathlon.model.Referee;
import triathlon.model.Result;

import java.util.List;

public interface IServices {
    Referee login(String username, IObserver client) throws ServiceException;
    List<TrialDTO> getTrialNames();
    List<ParticipantDTO> getParticipantsAndPoints() throws ServiceException;
    List<ParticipantDTO> filterParticipantsByTrial(Long idTrial);
    void addResult(Result result) throws ServiceException;
    void logout(Long idReferee, IObserver client) throws ServiceException;
    void sendFilter(Long idTrial);
}
